WIP: Introduce a new package named containerd

vikramhh opened the pull request:
1. Restructure code to make it easier to have multiple packages.
2. Introduce the containerd package
3. Introduce implicit links between a package and its source's
   names (a source for a given package should start with that
   package's name)
4. Handle multiple names and wildcards in package name i
   specification

Pending:

1. Specification and handling of Dependencies - future work.
2. Installation of the containerd service. This will be done
   in short order.

Signed-off-by: Vikram bir Singh <vikrambir.singh@docker.com>

Commit:
Introduce a new package named containerd

1. Restructure code to make it easier to have multiple packages.
2. Introduce the containerd package
3. Introduce implicit links between a package and its source's
   names (a source for a given package should start with that
   package's name)
4. Handle multiple names and wildcards in package name
   specification
5. Installation of the containerd service.
6. Removal of redundant/duplicated code and bug fixes for
   issues encountered while making changes.

Pending:

1. Specification and handling of Dependencies - future work.

Signed-off-by: Vikram bir Singh <vikrambir.singh@docker.com>

vikramhh commented:
This is a WIP - most of the debugging stuff will be removed along with some of the newly introduced unused variables [or they will be used in the next checkin].

I verified end to end scenarios around Find-Package and Install-Package with the provider.

StefanScherer commented on DockerMsftProvider.psm1:
```diff
@@ -21,12 +48,17 @@ $script:Installer_Extension = "zip"
 $script:dockerURL = "https://go.microsoft.com/fwlink/?LinkID=825636&clcid=0x409"
```
Looks like this variable is no longer used.

vikramhh replied:
Removed

StefanScherer reviewed (commented):
Only a duplicate variable so far.
In the meantime I tried to setup a test environment to test installation. I'll ask some questions about the artifacts in a different channel.

Commit:
Docker default install should install containerd

The default installation of Docker [without specifying
a version number] should lead to installation of the
containerd package as well.

Signed-off-by: vikrambirsingh <vikrambir.singh@docker.com>
